src/main/java/codeu/controller/AdminServlet.java

```diff
@@ -0,0 +1,142 @@
+   */
+  void setUserStore(UserStore userStore) {
+    this.userStore = userStore;
+  }
+  /**
+   * Sets the ConversationStore used by this servlet. This function provides a common setup method for use
+   * by the test framework or the servlet's init() function.
+   */
+  void setConversationStore(ConversationStore conversationStore) {
+    this.conversationStore = conversationStore;
+  }
+
+  /**
+   * Sets the MessageStore used by this servlet. This function provides a common setup method for
+   * use by the test framework or the servlet's init() function.
+   */
+  void setMessageStore(MessageStore messageStore) {
+    this.messageStore = messageStore;
+  }
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response)
+      throws IOException, ServletException {
+      String username = (String)request.getSession().getAttribute("user");
+
+      if(username == null) {
+        // user is not logged in, don't let them access the admin page
+        response.sendRedirect("/");
+        return;
+      }
+      List<String> adminUsernames = new ArrayList<>();
+      adminUsernames.add("admin");
+      adminUsernames.add("admin1");
+      adminUsernames.add("admin2");
+      adminUsernames.add("admin3");
+
+
+      User user = userStore.getUser(username);
+
+      if (adminUsernames.contains(username)){
+        User userAdmin = userStore.getUser(username);
+        user.setIsAdmin(true);
+      }
+
+      if(user == null) {
+        // user was not found, don't let them access the admin page
+        response.sendRedirect("/");
+        return;
+      }
+      if(user.getIsAdmin() == false) {
+        response.sendRedirect("/");
+        return;
+      }
+      /* an attempt to grab information from the stores to display on the page
+      if the user is admin*/
+      if(!adminUsernames.contains(username)){
+        adminUsernames.add(username);
+      }
+        int numOfUsers = userStore.amountOfUsers();
+        int numOfConvos = conversationStore.amountOfConnversations();
+        int numOfMessages = messageStore.amountOfMessages();
+
+        request.setAttribute("numOfUsers", numOfUsers);
+        request.setAttribute("numOfConvos", numOfConvos);
+        request.setAttribute("numOfMessages", numOfMessages);
+
+        List<User> userList = userStore.getAllUsers();
+        String newestUser ="";
+        String mostActiveUser = null;
+        int numOfAdmins = adminUsernames.size();
+        if (userList.size()>0){
+            newestUser = userList.get(numOfUsers-1).getName();
+        }
+
+        for(int i =0; i< numOfUsers; i++){
+          User userInfo = userList.get(i);
+          UUID userId = userInfo.getId();
+          int userMessagesListSize = messageStore.getMessagesFromAuthor(userId).size();
+          int mostMessages=0;
+          if (userMessagesListSize > mostMessages){
+            mostMessages = messageStore.getMessagesFromAuthor(userId).size();
```

Comment:
you can just set mostMessages = userMessagesListSize, instead of calling the methods again.